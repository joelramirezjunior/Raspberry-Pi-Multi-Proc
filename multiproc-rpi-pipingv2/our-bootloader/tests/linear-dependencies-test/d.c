#include "rpi.h"

int do_stuff(int a) {
    unsigned long long i = 0;
    printk("     doing stuff... should take a bit\n");
    while (0 != i++) {
    	a++;
    }
    return 140 * a + 0xe;
}

void notmain(void) {
    printk("continuing linear dependencies test.\n");
    printk("starting d, which is depended on by NOTHING. ----------------------\n");
    //todo: int d = READ FROM PIPE;
    int d = 10;
    printk("from c, got output: ---%d---\n", d);
    trace("from c, got output: ---%d---\n", d);
    printk("done with d. --------------------------------------\n");
    printk("done with all linear dependencies. you should see the correct value above, showing everything worked.\n");
}

