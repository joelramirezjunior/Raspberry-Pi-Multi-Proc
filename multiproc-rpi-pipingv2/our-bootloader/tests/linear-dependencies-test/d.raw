FOUND: </dev/cu.SLAB_USBtoUART>
opened tty port </dev/cu.SLAB_USBtoUART>.
pi-install: tty-usb=</dev/cu.SLAB_USBtoUART> program=<./linear-dependencies-test/d.bin> about to boot
TRACE:simple_boot: sending 3048 bytes, crc32=2c7b0c1
waiting for a start
putting code
pi sent print: <UART Teddy Zhang: success: Received the program!>
bootloader: Done.
listening on ttyusb=</dev/cu.SLAB_USBtoUART>
continuing linear dependencies test.
starting d, which is depended on by NOTHING. ----------------------
from c, got output: ---10---
TRACE:notmain:from c, got output: ---10---
done with d. --------------------------------------
done with all linear dependencies. you should see the correct value above, showing everything worked.
DONE!!!

Saw done

bootloader: pi exited.  cleaning up
