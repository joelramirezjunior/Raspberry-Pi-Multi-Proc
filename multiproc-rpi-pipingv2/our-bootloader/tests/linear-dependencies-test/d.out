TRACE:notmain:from c, got output: ---10---
